docs/api/input.md

```diff
@@ -24,341 +17,110 @@ import EncapsulationPill from '@components/page/api/EncapsulationPill';
 
 <EncapsulationPill type="scoped" />
 
-<h2 className="table-of-contents__title">Contents</h2>
-
-<TOCInline
-  toc={toc}
-  maxHeadingLevel={2}
-/>
-
-
 
 The input component is a wrapper to the HTML input element with custom styling and additional functionality. It accepts most of the same properties as the HTML input, but works great on desktop devices and integrates with the keyboard on mobile devices.
 
-It is meant for text `type` inputs only, such as `"text"`, `"password"`, `"email"`, `"number"`, `"search"`, `"tel"`, and `"url"`. It supports all standard text input events including keyup, keydown, keypress, and more.
-
-## Interfaces
-
-### InputChangeEventDetail
-
-```typescript
-interface InputChangeEventDetail {
-  value: string | undefined | null;
-}
-```
-
-### InputCustomEvent
-
-While not required, this interface can be used in place of the `CustomEvent` interface for stronger typing with Ionic events emitted from this component.
-
-```typescript
-interface InputCustomEvent extends CustomEvent {
-  detail: InputChangeEventDetail;
-  target: HTMLIonInputElement;
-}
-```
 
+## Basic Usage
 
+import Basic from '@site/static/usage/input/basic/index.md';
 
-## Usage
+<Basic />
 
-<Tabs groupId="framework" defaultValue="angular" values={[{ value: 'angular', label: 'Angular' }, { value: 'javascript', label: 'Javascript' }, { value: 'react', label: 'React' }, { value: 'stencil', label: 'Stencil' }, { value: 'vue', label: 'Vue' }]}>
 
-<TabItem value="angular">
+## Types
 
-```html
-<!-- Default Input -->
-<ion-input></ion-input>
+The input component is meant for text type inputs only, such as `"text"`, `"password"`, `"email"`, `"number"`, `"search"`, `"tel"`, and `"url"`. It supports all standard text input events including keyup, keydown, keypress, and more. The default `type` is `"text"`.
 
-<!-- Input with value -->
-<ion-input value="custom"></ion-input>
+import Types from '@site/static/usage/input/types/index.md';
 
-<!-- Input with placeholder -->
-<ion-input placeholder="Enter Input"></ion-input>
+<Types />
 
-<!-- Input with clear button when there is a value -->
-<ion-input clearInput value="clear me"></ion-input>
 
-<!-- Number type input -->
-<ion-input type="number" value="333"></ion-input>
+## Label Positioning
 
-<!-- Disabled input -->
-<ion-input value="Disabled" disabled></ion-input>
+Labels will take up the width of their content by default. This positioning can be changed to be a fixed width, stacked, or floating label.
 
-<!-- Readonly input -->
-<ion-input value="Readonly" readonly></ion-input>
+import Labels from '@site/static/usage/input/labels/index.md';
 
-<!-- Inputs with labels -->
-<ion-item>
-  <ion-label>Default Label</ion-label>
-  <ion-input></ion-input>
-</ion-item>
+<Labels />
```

Comment:
I thought about just reusing the input labels playground here: https://ionicframework.com/docs/api/label#input-labels

But I decided against it because that one also included things like toggle and checkbox. I figured it didn't hurt to add it in since we already the label documentation is already a bit buried under the "item" category and the label positioning like this really only works well with input.

Reply:
Makes sense! I'm cool with keeping it.